### DISCUSSION TITLE
readRSSI value update frequency?

### DISCUSSION THREAD

**Discussion by rithvik-doshi** [innoveit/react-native-ble-manager#1135] [Q&A]
Even if I call readRSSI value every 100ms, it seems like the RSSI value only changes values every half second or so. Is this functionality intended? Code for context

```javascript
useEffect(() => {
    const interval = setInterval(() => {
      if (isConnected) {
        scanIds.map((item) => retrieveRssi(item));
      }
    }, 100);
    return () => clearInterval(interval);
  }, [isConnected, scanIds, diff1, diff2]);

const retrieveRssi = (peripheralId) => {
    BleManager.readRSSI(peripheralId)
      .then((rssi) => {
        index = scanIds.indexOf(peripheralId);
        if (index == 0) {
          setDiff1(rssi);
        } else {
          setDiff2(rssi);
        }
      })
      .then(console.log(diff1, diff2))
      .catch((error) => {
        // Failure code
        console.log(error);
      });
  };
```

**Comment by marcosinigaglia** [comment]
Why do you need the rssi value so frequently?

**Reply by rithvik-doshi** [reply]
I want to get the rssi of two bluetooth peripherals with low latency so that I can trigger a notification when one is greater than the other. This works with my current setup, but even then it takes a second or two before the notification is triggered, whereas I'd ideally like it to trigger in under a second.

**Comment by marcosinigaglia** [comment]
Probably the best way is to scan and retrieves the rssi in the advertisings
